src/ILLink.RoslynAnalyzer/DynamicallyAccessedMembersAnalyzer.cs

```diff
@@ -114,7 +114,8 @@ static SingleValue GetTypeValueNodeFromGenericArgument (ITypeSymbol type)
 				// That said we only use it to perform the dynamically accessed members checks and for that purpose treating it as System.Type is perfectly valid.
 				SymbolKind.NamedType => new SystemTypeValue ((INamedTypeSymbol) type),
 				SymbolKind.ErrorType => UnknownValue.Instance,
-				// What about things like ArrayType or PointerType and so on. Linker treats these as "named types" since it can resolve them to concrete type
+				SymbolKind.ArrayType => new ArrayValue (GetTypeValueNodeFromGenericArgument (((IArrayTypeSymbol) type).ElementType)),
+				// What about things like PointerType and so on. Linker treats these as "named types" since it can resolve them to concrete type
```

Comment:
Since we're here, we should handle these as well - linker resolves pointers and alike to their base element type - so for example "foo *" is resolved to "foo" and then tracked like that. I don't actually know if that is correct from semantics point of view (as in, what will reflection do). So it would be good to figure out the reflection behavior and potentially file an issue against linker as well.

Reply:
I don't think pointers can be used as generic arguments, at least in C# - do we even need to handle these?

Reply:
That's true, at least for now - I vaguely remember that there's some discussion to allow refs there (reflection could really use this to avoid boxing I think), but I could be wrong.